packages/@azure/servicebus/data-plane/test/testUtils.ts

```diff
@@ -146,11 +146,16 @@ async function recreateSubscription(
   subscriptionName: string,
   parameters: SBSubscription
 ): Promise<void> {
-  const env = await getEnvVars();
+  const env = getEnvVars();
   await msRestNodeAuth
     .loginWithServicePrincipalSecret(env.clientId, env.secret, env.tenantId)
     .then(async (creds) => {
       const client = await new ServiceBusManagementClient(creds, env.subscriptionId);
+      /*
+        Unlike Queues/Topics, there is no need to delete the subscription because
+        `recreateTopic` is called before `recreateSubscription` which would
+        delete the topic and the subscriptions, creates a new topic.
```

Comment:
which would delete the topic and the subscriptions **before creating** a new topic.